### DISCUSSION TITLE
Intermintent lose of motion events

### DISCUSSION THREAD

**Discussion by jeffgraham-hub** [tsightler/ring-mqtt#916] [General Discussion]
[https://github.com/tsightler/ring-mqtt/discussions/896#discussioncomment-10919074](url)
> But, assuming it happens again, we'd need logs, but also, ideally, a login to the container to check the established network connections and, in a perfect world, a network trace while the problem is happening. 
> 
> If you can do all of this great, if not, I'll probably need to build custom version of ring-client-api and push-receiver that instrument some of this so that we have any chance to figure out what is happening. If you are willing to commit some time over the next months, I'd say we should open a new issue and work on it long term.
>

After a little searching this is what I have come up with based upon your request.
1. Logs
  I have logging setup via ssh. I ran the following command with the output dumping to a file. 
```
docker logs -f addon_a98c085d_ring_mqtt
```

2. Check established network connections.
    Is this as simple as the output of nestat -a or are you requesting something different?
```
docker exec -it addon_a98c085d_ring_mqtt bash
netstat -a
```
3. Network trace
   This one I'm not sure on. I tried tcpdump but it isn't included in the image. If that is what you are wanting I will search how to install and use it.

**Comment by tsightler** [comment]
Let's wait and see what happens before the dump.  It's easy to install tcpdump into the container if required "apk install tcpdump" but perhaps it won't be required if there is something interesting in the logs.

**Comment by jeffgraham-hub** [comment]
I lost motion events last night. I believe this instance can be attributed to lose of internet services. I experienced an outage for approximately an hour last night, after which everything appears to have recovered except push notifications. Below is a snippet of the logs that references push-receiver during the outage. I don't recall any network issues during the previous lose of motion events but I can't rule that out. The complete log is 80 MB so I emailed you the timeframe (23:30 - 4:53) surrounding the network outage. I can share the complete log via Goggle Drive or similar if you would like. Also, if there is something else you want me to check before restarting ring-mqtt let me know.

```

2024-10-13T23:49:10.090Z ring-mqtt WARNING - Unhandled Promise Rejection
2024-10-13T23:49:10.095Z ring-mqtt fetch failed
2024-10-13T23:49:10.096Z ring-mqtt TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async retry (/app/ring-mqtt/node_modules/@eneris/push-receiver/dist/utils/request.js:19:16)
    at async checkIn (/app/ring-mqtt/node_modules/@eneris/push-receiver/dist/gcm.js:24:25)
    at async PushReceiver.registerIfNeeded (/app/ring-mqtt/node_modules/@eneris/push-receiver/dist/client.js:147:13)
    at async PushReceiver.connect (/app/ring-mqtt/node_modules/@eneris/push-receiver/dist/client.js:91:9)
2024-10-14T00:46:47.902Z ring-mqtt Websocket for location id xxxxxxxx is connected


```

**Reply by tsightler** [reply]
Thanks @jeffgraham-hub!  I'm traveling this week so could be some delay to dig into deeply but this looks like something we need to get fixed in push-receiver.  I'll open an issue with upstream and also see if I can hack up a fix.

**Reply by tsightler** [reply]
Opened an issue with upstream here:  https://github.com/Eneris/push-receiver/issues/28

**Reply by tsightler** [reply]
Thanks again @jeffgraham-hub for the initial logging data here.  We've managed to get a changeset implemented upstream that passes the same network issue without crashing.  A temporary fix is in the dev branch of ring-mqtt.  As soon as upstream publishes a new package that includes this fix I'll push out an update.  Hopefully that will improve the reliability of push notifications going forward as, with this new fix, I've not been able to reproduce failures no matter what I throw at it.

**Reply by jeffgraham-hub** [reply]
@tsightler ...Just as a follow up I can confirm the only issue I have had is related  to loss of network connectivity. I appreciate you taking the time to investigate the issue.
